implementations/java-json-schema/BowtieJsonSchema.java

```diff
@@ -115,18 +120,42 @@ private void dialect(JsonNode node) throws JsonProcessingException {
       DialectRequest.class
     );
 
-    // well, setting default dialect shouldn't be necessary but why not
-    if (SpecificationVersion.DRAFT2020_12.getId().equals(dialectRequest.dialect())) {
-        validatorFactory.withDefaultDialect(new Dialects.Draft2020Dialect());
-    } else if (SpecificationVersion.DRAFT2019_09.getId().equals(dialectRequest.dialect())) {
-        validatorFactory.withDefaultDialect(new Dialects.Draft2019Dialect());
-    } else if (SpecificationVersion.DRAFT7.getId().equals(dialectRequest.dialect())) {
-      validatorFactory.withDefaultDialect(new Dialects.Draft7Dialect());
+    ConcurrentHashMap<String, Dialect> dialectsMap = Arrays.stream(Dialects.class.getClasses())
+            .filter(Dialect.class::isAssignableFrom)
```

Comment:
I think this filter always evaluates to `false` - that's my bad (I also forgot about actually instantiating dialect objects but I see you figured it out already :). And it seems that filtering classes to be of a right type is a bit trick if you have to operate on `Class<?>` instances. So I would just get rid of the whole filtering, I don't think we gain much from it (worst case scenario it will throw exception and at least there will some indication what went wrong)

Reply:
> I think this filter always evaluates to false 

Well, turns out I don't know what I'm saying, it's all good, don't mind the whole previous comment :)